exporter/datadogexporter/config.go

```diff
@@ -479,6 +490,23 @@ func (c *Config) Validate() error {
 		return err
 	}
 
+	if !isLogsAgentExporterEnabled() {
+		if c.Logs.UseCompression {
+			return fmt.Errorf("logs::use_compression is not valid when the exporter.datadogexporter.logsagentexporter feature gate is disabled")
+		}
+		if c.Logs.CompressionLevel != 0 {
+			return fmt.Errorf("logs::compression_level is not valid when the exporter.datadogexporter.logsagentexporter feature gate is disabled")
+		}
+		if c.Logs.BatchWait != 0 {
+			return fmt.Errorf("logs::batch_wait is not valid when the exporter.datadogexporter.logsagentexporter feature gate is disabled")
+		}
+	}
+	if isLogsAgentExporterEnabled() {
+		if c.Logs.DumpPayloads {
+			return fmt.Errorf("logs::dump_payloads is not valid when the exporter.datadogexporter.logsagentexporter feature gate is enabled")
+		}
+	}
+
```

Comment:
nit: This won't be able to identify if the user has set these settings explicitly to the default value. Instead of checking here, we can check on the `Unmarshal` method below by checking e.g. `configMap.IsSet("logs::dump_payloads")`.

This should also simplify the code (you can just loop through a slice of all the options and check if any `IsSet`)